Sources/StructuralCore/Structural.swift

```diff
@@ -25,57 +27,55 @@ public protocol Structural {
 }
 
 /// Structural representation of a Swift struct.
-public struct StructuralStruct<Properties> {
-    public var type: Any.Type?
+public struct StructuralStruct<BaseType, Properties> {
+    public var type: BaseType.Type?
```

Comment:
Why is `type` `Optional<BaseType.Type>` and not simply `BaseType.Type`? (Note: I realize this was the case before your change too.)

Reply:
If you look at implementation of `Zero` we need to be able to create an instance `StructuralStruct` based on known type parameters only.

Reply:
But don't we now have `BaseType`? Couldn't we do something like:

```swift
extension StructuralStruct: Zero where Properties: Zero {
    public static var zero: Self {
        return StructuralStruct(type: BaseType.self, Properties.zero)
    }
}
```